src/internal/src/linux/AudioDestinationLinux.cpp

```diff
@@ -0,0 +1,168 @@
+  for (size_t i = 0; i < n; i++) {
+    RtAudio::DeviceInfo info(audio.getDeviceInfo(i));
+    if (info.isDefaultOutput) {
+      return info.outputChannels;
+    }
+  }
+  return 2;
+}
+
+const float kLowThreshold = -1.0f;
+const float kHighThreshold = 1.0f;
+
+AudioDestination * AudioDestination::MakePlatformAudioDestination(AudioIOCallback & callback, unsigned numberOfOutputChannels, float sampleRate)
+{
+    return new AudioDestinationLinux(callback, numberOfOutputChannels, sampleRate);
+}
+
+unsigned long AudioDestination::maxChannelCount()
+{
+    return NumDefaultOutputChannels();
+}
+
+AudioDestinationLinux::AudioDestinationLinux(AudioIOCallback & callback, unsigned numberOfOutputChannels, float sampleRate) : m_callback(callback)
+{
+    m_sampleRate = sampleRate;
+    m_renderBus.setSampleRate(m_sampleRate);
+    dac.reset(new RtAudio()); // XXX
+    configure();
+}
+
+AudioDestinationLinux::~AudioDestinationLinux()
+{
+    dac.release(); // XXX
+    /* if (dac.isStreamOpen())
+        dac.closeStream(); */
+}
+
+void AudioDestinationLinux::configure()
+{
+    if (dac->getDeviceCount() < 1)
+    {
+        LOG_ERROR("No audio devices available");
+    }
+
+    dac->showWarnings(true);
+
+    RtAudio::StreamParameters outputParams;
+    outputParams.deviceId = dac->getDefaultOutputDevice();
+    outputParams.nChannels = channelCount();
+    outputParams.firstChannel = 0;
+
+    auto deviceInfo = dac->getDeviceInfo(outputParams.deviceId);
+    LOG("Using Default Audio Device: %s", deviceInfo.name.c_str());
+
+    RtAudio::StreamParameters inputParams;
+    inputParams.deviceId = dac->getDefaultInputDevice();
+    inputParams.nChannels = 1;
+    inputParams.firstChannel = 0;
+
+    unsigned int bufferFrames = AudioNode::ProcessingSizeInFrames;
+
+    RtAudio::StreamOptions options;
+    options.flags |= RTAUDIO_NONINTERLEAVED;
+
+    try
+    {
+        dac->openStream(&outputParams, &inputParams, RTAUDIO_FLOAT32, (unsigned int) m_sampleRate, &bufferFrames, &outputCallback, this, &options);
+    }
+    catch (RtAudioError & e)
+    {
+        e.printMessage();
+    }
+}
+
+void AudioDestinationLinux::start()
+{
+    try
+    {
+        dac->startStream();
+        m_isPlaying = true;
+    }
+    catch (RtAudioError & e)
+    {
+        e.printMessage();
+    }
+}
+
+void AudioDestinationLinux::stop()
+{
+    try
+    {
+        // dac->stopStream(); // XXX
+        m_isPlaying = false;
+    }
+    catch (RtAudioError & e)
+    {
+        e.printMessage();
+    }
+}
+
+// Pulls on our provider to get rendered audio stream.
+void AudioDestinationLinux::render(int numberOfFrames, void * outputBuffer, void * inputBuffer)
+{
+    float *myOutputBufferOfFloats = (float*) outputBuffer;
+    float *myInputBufferOfFloats = (float*) inputBuffer;
+
+    // Inform bus to use an externally allocated buffer from rtaudio
+    if (m_renderBus.isFirstTime())
+    {
+        m_renderBus.setChannelMemory(0, myOutputBufferOfFloats, numberOfFrames);
```

Comment:
Since there's support for multi-channel, don't we need to loop `m_renderBus.numerOfChannels()` for this?